src/commands/operator/recover.ts

```diff
@@ -126,23 +147,67 @@ export default class Recover extends OperatorJobAwareCommand {
     await this.networkMonitor.run(true)
     CliUx.ux.action.stop()
 
-    CliUx.ux.action.start('Retrieving transaction details from ' + network + ' network')
-    const transaction = await this.networkMonitor.getTransaction({
-      transactionHash: tx,
-      network,
-      canFail: true,
-      attempts: 30,
-      interval: 500,
-    })
-    CliUx.ux.action.stop()
+    let txArray = []
+    // If network and tx flags are provided, construct an array of one element
+    if(!!flags.network || !!flags.tx) {
+      const flagNetwork: string = await checkOptionFlag(
+        supportedNetworks,
+        flags.network,
+        'Select the network to extract transaction details from',
+      )
+
+      const flagTx: string = await checkTransactionHashFlag(
+        flags.tx,
+        'Enter the hash of transaction from which to extract recovery data from',
+      )
+
+      txArray = [{ sourceChainId: networkToChainId[flagNetwork], sourceTx: flagTx }]
 
-    if (transaction === null) {
-      this.networkMonitor.structuredLog(network, 'Could not retrieve the transaction')
-      // eslint-disable no-process-exit, unicorn/no-process-exit
-      this.exit()
     } else {
-      await this.processTransaction(network, transaction)
+      if (!(await fs.pathExists(flags.file as string))) {
+        this.error(`Problem reading ${flags.file}`)
+      }
+
+      let incompleteJobs: IncompleteJobs[] = []
+      try {
+        incompleteJobs = (await fs.readJson(flags.file as string)) as IncompleteJobs[]
+        //TODO: add validation of the file
+      } catch {
+        this.error(`One or more lines are an invalid Incomplete jobs JSON object`)
+      }
+
+      txArray = incompleteJobs.map(item => {
+        return {
+          sourceChainId: item.source_chain_id,
```

Comment:
Oh looks like it's formatted with snake case to match the way it is in the file? Where does the file data come from? db?

Reply:
@alexanderattar Yea its snake case because its from the DB.